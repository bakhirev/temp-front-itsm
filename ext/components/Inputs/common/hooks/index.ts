export * from './useSuffixHook';
export * from './useTooltipHook';
