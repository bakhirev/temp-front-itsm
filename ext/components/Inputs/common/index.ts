export * from './interface';
export * from './hooks';
export * from './utils';
export * from './helpers';
export * from './types';
