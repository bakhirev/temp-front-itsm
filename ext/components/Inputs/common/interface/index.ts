export * from './iInputDefaultProps';
export * from './iStyledIconDefaultProps';
