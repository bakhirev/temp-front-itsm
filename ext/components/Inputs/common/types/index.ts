/** Типы для компонентов инпут */

export type Size = 'big' | 'small' | 'micro';
export type Status = 'success' | 'error' | 'default';
export type Mask = 'card' | 'phone';
export type Type = 'text' | 'password' | 'suggest' | 'number' | 'range' | 'informer';
