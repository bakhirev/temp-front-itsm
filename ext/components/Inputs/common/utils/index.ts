export * from './formatValue';
export * from './utils-style';
export * from './getIconSize';
