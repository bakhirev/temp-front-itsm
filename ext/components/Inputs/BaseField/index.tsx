import React, { useState, Dispatch, SetStateAction, ReactNode, FC } from 'react';

import { createTooltipHOC } from '../../TooltipHOC';
import { Body2Short } from '../../Typography';
import { IInputDefaultProps, useTooltipHook } from '../common';
import type { Type } from '../common/types';
import { DEFAULT_WIDTH } from '../constants';

import { InputSkeleton } from './InputSkeleton';
import { Label, Wrapper, Separator, StyledAdditionalText } from './StyledComponents';

export interface IInputProps extends IInputDefaultProps {
  /** Коллбэк на клик */
  onClick?: (event: any) => void | undefined;
  /** иконки из Figma , которые прокидываются в инпут */
  iconServices?: ReactNode;
  /** Отображение подсказки при переполнении инпута. По умолчанию false */
  withTooltip?: boolean;
  /** Тип инпута */
  type?: Type;
  /** Активна ли иконка в инпуте */
  iconIsActive?: boolean;
  /** Функция для переключения active icon */
  setIconIsActive?: Dispatch<SetStateAction<boolean>>;
  /** Использование числового инпута */
  number?: boolean;
  /** Использование Input Range */
  range?: boolean;
  /** Активен ли информер в инпуте */
  activeInformer?: boolean;
  /** Значение инпута */
  value?: any;
  /** Флаг ввода дробной части */
  withCoins?: boolean;
  phoneCountry?: boolean;
  countryIcon?: any;
  selectIcon?: any;
  onCountrySelectOpen?: Function;
  countrySelectComponent?: any;
  onCountrySelect?: Function;
  /** Дефолтное value */
  defaultValue?: any;
}

export const Input: FC<IInputProps> = ({
  disabled,
  label,
  additionalText,
  status = 'default',
  className,
  dataTestId,
  withTooltip = false,
  value = '',
  width = DEFAULT_WIDTH,
  refInput,
  tooltipContainer,
  ...props
}) => {
  const [focused, setFocused] = useState(false);
  const { tooltip, ref } = useTooltipHook(refInput, withTooltip, value);

  const Input = tooltip && !focused ? createTooltipHOC(InputSkeleton) : InputSkeleton;
  const inputElement = (
    <Input
      tooltip={value}
      container={tooltipContainer}
      display="block"
      disabled={disabled}
      focused={focused}
      setFocused={setFocused}
      status={status}
      value={value}
      ref={ref}
      width={width}
      {...props}
    />
  );

  return (
    <Wrapper
      disabled={disabled}
      focused={focused}
      className={className}
      data-test-id={dataTestId}
      width={width}
    >
      <Label>
        {label && (
          <>
            <Body2Short>{label}</Body2Short>
            <Separator />
          </>
        )}
        {inputElement}
      </Label>
      {additionalText && (
        <StyledAdditionalText disabled={disabled} status={status} focused={focused}>
          <Body2Short>{additionalText}</Body2Short>
        </StyledAdditionalText>
      )}
    </Wrapper>
  );
};
