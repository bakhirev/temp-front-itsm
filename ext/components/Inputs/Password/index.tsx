import React, { useState, ReactNode, FC } from 'react';

import { Input } from '../BaseField';
import { getIconSize } from '../common';
import type { IInputDefaultProps } from '../common/interface';
import type { Type } from '../common/types';

import { StyledEyeSolid } from './StyledEyeSolid';
import { StyledEyeCloseOutline } from './StyledEyeCloseOutline';

export interface IInputPassword extends IInputDefaultProps {
  /** Значение инпута */
  value?: string | undefined;
  /** Дополнительная иконка */
  icon?: ReactNode;
}

export const InputPassword: FC<IInputPassword> = ({
  value = '',
  disabled,
  status,
  size = 'big',
  onChange,
  ...props
}) => {
  const [showPassword, setShow] = useState(false);
  const [typeInput, setTypeInput] = useState<Type>('password');

  const handleTogglePassword = () => {
    !disabled && setShow(!showPassword);
    !disabled && showPassword ? setTypeInput('password') : setTypeInput('text');
  };

  const handleClickClearingInput = (event: any) => {
    status === 'error' && onChange && onChange(event, '');
  };

  const iconSize = getIconSize(size);
  const iconServices = (
    <>
      {!showPassword ? (
        <StyledEyeSolid
          onMouseDown={handleTogglePassword}
          width={iconSize}
          height={iconSize}
          disabled={disabled}
        />
      ) : (
        <StyledEyeCloseOutline
          onMouseDown={handleTogglePassword}
          width={iconSize}
          height={iconSize}
          disabled={disabled}
        />
      )}
    </>
  );

  return (
    <Input
      type={typeInput}
      size={size}
      disabled={disabled}
      status={status}
      iconServices={iconServices}
      onClick={(event: any) => handleClickClearingInput && handleClickClearingInput(event)}
      onChange={onChange}
      value={value}
      {...props}
    />
  );
};
