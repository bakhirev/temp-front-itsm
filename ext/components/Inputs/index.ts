import { TimePicker } from '../TimePicker';

import { InputText } from './Text';
import { InputPassword } from './Password';
import { InputSuggest } from './Suggest';
import { InputInformer } from './Informer';
import { InputNumber } from './Number';
import { InputMask } from './Mask';
import { InputRange } from './InputRange';
import { InputPhoneCountry } from './PhoneCountry';
export { IInputProps } from './BaseField';
export { IInputMaskProps } from './Mask';
export type { IOptionSuggest } from './Suggest';

export const Input = {
  Text: InputText,
  Password: InputPassword,
  Number: InputNumber,
  Mask: InputMask,
  Suggest: InputSuggest,
  Range: InputRange,
  Informer: InputInformer,
  PhoneCountry: InputPhoneCountry,
  Time: TimePicker,
};
