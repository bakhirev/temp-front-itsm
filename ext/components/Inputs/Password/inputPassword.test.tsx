import { runCommonInputTest, runCommonInputOnChangeTest } from '../common/test';
import { Input } from '../../Inputs';

jest.mock('../../common', () => ({
  DEFAULT_THEME: {},
  Z_INDEXES: {
    INFORMER: 777,
  },
}));

describe('Input.Password', () => {
  runCommonInputTest(Input.Password);

  runCommonInputOnChangeTest(Input.Password, '123qwerty', '123qwerty');
});
