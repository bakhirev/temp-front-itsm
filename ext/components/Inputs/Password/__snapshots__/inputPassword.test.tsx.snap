// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input.Password should render component 1`] = `
<Component
  iconServices={
    <React.Fragment>
      <ForwardRef(styled.div)
        height={24}
        onMouseDown={[Function]}
        theme={Object {}}
        width={24}
      />
    </React.Fragment>
  }
  onChange={[Function]}
  onClick={[Function]}
  size="big"
  type="password"
  value=""
/>
`;

exports[`Input.Password should render component with clearable field, icon and name 1`] = `
<Component
  clearable={true}
  icon={<div />}
  iconServices={
    <React.Fragment>
      <ForwardRef(styled.div)
        height={20}
        onMouseDown={[Function]}
        theme={Object {}}
        width={20}
      />
    </React.Fragment>
  }
  inputName="test"
  label="Введите ваше имя"
  onChange={[Function]}
  onClick={[Function]}
  size="micro"
  type="password"
  value=""
/>
`;

exports[`Input.Password should render component with error status 1`] = `
<Component
  additionalText="Ошибка"
  iconServices={
    <React.Fragment>
      <ForwardRef(styled.div)
        height={24}
        onMouseDown={[Function]}
        theme={Object {}}
        width={24}
      />
    </React.Fragment>
  }
  label=""
  onChange={[Function]}
  onClick={[Function]}
  size="big"
  status="error"
  type="password"
  value=""
/>
`;

exports[`Input.Password should render component with success status 1`] = `
<Component
  additionalText="Успех"
  iconServices={
    <React.Fragment>
      <ForwardRef(styled.div)
        height={24}
        onMouseDown={[Function]}
        theme={Object {}}
        width={24}
      />
    </React.Fragment>
  }
  label="Введите ваше имя"
  onChange={[Function]}
  onClick={[Function]}
  size="big"
  status="success"
  type="password"
  value=""
/>
`;
